src/mesh_analysis.rs

```diff
@@ -223,7 +223,7 @@ pub fn are_visually_identical(g_1: &Geometry, g_2: &Geometry) -> bool {
     g_1.faces().len() == g_2.faces().len()
         && g_1.vertices().len() == g_2.vertices().len()
         && g_1.normals().len() == g_2.normals().len()
-        && unpacked_faces_1.all(|f| unpacked_faces_2.any(|g| f == g))
+        && unpacked_faces_1.all(|f| unpacked_faces_2.clone().any(|g| f == g))
```

Comment:
Does this means the tests for this function were insufficient? Maybe we should add more tests that cover the bug that was just fixed? Ideally separate this out of the PR and let's do it first.

Reply:
not sure why do the original tests pass but they shoudn't. i'll add more tests catching this bug.

Reply:
tests now prove it works (wouldn't pass with the former version)